File: ApemanPlugin/src/main/kotlin/apeman_core/features_extraction/calculators/method/NumInvocationMethodCalculator.java
package apeman_core.features_extraction.calculators.method;

import apeman_core.base_entities.FeatureType;
import apeman_core.pipes.CandidateWithFeatures;
import com.intellij.psi.JavaRecursiveElementVisitor;
import com.intellij.psi.PsiMethodCallExpression;
import com.intellij.psi.PsiNewExpression;

import java.util.ArrayList;

public class NumInvocationMethodCalculator extends NumSimpleElementMethodCalculator {

    public NumInvocationMethodCalculator(ArrayList<CandidateWithFeatures> candidates) {
        super(candidates, FeatureType.CON_INVOCATION);
    }

    @Override
    public JavaRecursiveElementVisitor createVisitor() {
        return new Visitor();
    }

    private class Visitor extends NumSimpleElementMethodCalculator.Visitor {

        @Override
        public void visitMethodCallExpression(PsiMethodCallExpression expression) {
            super.visitMethodCallExpression(expression);
            elementsCounter++;
        }

        @Override
        public void visitNewExpression(PsiNewExpression exp) {
            super.visitNewExpression(exp);
            if (exp.getArrayDimensions().length == 0 &&
                    exp.getArrayInitializer() == null) {
                elementsCounter++;
            }
        }
    }
}

File: ApemanPlugin/src/main/kotlin/apeman_core/features_extraction/calculators/method/NumInvocationMethodCalculator.kt
package apeman_core.features_extraction.calculators.method

import apeman_core.base_entities.ExtractionCandidate
import apeman_core.base_entities.FeatureType
import com.intellij.psi.PsiMethodCallExpression

class NumInvocationMethodCalculator(candidates: List<ExtractionCandidate>
) : NumSimpleElementMethodCalculator(candidates, FeatureType.CON_INVOCATION) {

    override fun createVisitor() = Visitor()

    inner class Visitor : NumSimpleElementMethodCalculator.Visitor() {

        override fun visitMethodCallExpression(expression: PsiMethodCallExpression) {
            super.visitMethodCallExpression(expression)
            elementsCounter++
        }

//        override fun visitNewExpression(exp: PsiNewExpression) {
//            super.visitNewExpression(exp)
//            if (exp.arrayDimensions.isEmpty() && exp.arrayInitializer == null) {
//                elementsCounter++
//            }
//        }

//        override fun visitTypeCastExpression(expression: PsiTypeCastExpression?) {
//            super.visitTypeCastExpression(expression)
//            elementsCounter++
//        }
    }
}
